### PR TITLE
feat: support cron expression in sync api

### PR DESCRIPTION
**Pull request by afthabvp**
## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] New Connector (Destination or Source Connector)
- [x] Breaking change (fix or feature that would impact existing functionality)
- [ ] Styling change
- [ ] Documentation update
- [ ] Refactoring
- [ ] Chore

## Checklist:

- [x] Ensure a branch name is prefixed with `feature`, `bugfix`, `hotfix`, `release`, `style` or `chore` followed by `/` and branch name e.g `feature/add-salesforce-connector`
- [x] Added unit tests for the changes made (if required)
- [ ] Have you made sure the commit messages meets the guidelines?
- [ ] Added relevant screenshots for the changes
- [ ] Have you tested the changes on local/staging?
- [ ] Added the new connector in rollout.rb
- [ ] Have you updated the version number of the gem?
- [ ] Have you ensured that your changes for new connector are documented in the [docs repo](https://github.com/Multiwoven/docs) or relevant documentation files?
- [ ] Have you updated the run time dependency in multiwoven-integrations.gemspec if new gems are added
- [ ] Have you made sure the code you have written follows the best practises to the best of your knowledge?

### CODE REVIEW

File: server/app/controllers/api/v1/syncs_controller.rb [hunk 1]
```diff
@@ -83,6 +85,20 @@ def set_sync
         @sync = current_workspace.syncs.find(params[:id])
       end
 
+      def modify_sync_params
```
Comment (subintp) [inline server/app/controllers/api/v1/syncs_controller.rb:88]: Why do we have to do this ?

File: server/app/models/sync.rb [hunk 2]
```diff
@@ -94,7 +98,17 @@ def set_defaults
     self.status ||= self.class.aasm.initial_state.to_s
   end
 
+  def interval_schedule?
```
Comment (subintp) [inline server/app/models/sync.rb:101]: these methods will be automatically created for enums. No need to add these methods ?

**Review by subintp — APPROVED** [review]

File: server/app/controllers/api/v1/syncs_controller.rb (hunk 1 above)
Comment (afthabvp) [reply server/app/controllers/api/v1/syncs_controller.rb:88]: It clears out irrelevant parameters before saving them to the database

File: server/app/models/sync.rb (hunk 2 above)
Comment (afthabvp) [reply server/app/models/sync.rb:101]: interval_schedule? method interval_schedule? checks whether the schedule_type is equal to "interval" and returns a boolean value accordingly.
 any suggestion for this?

**Review by subintp — APPROVED** [review]

**Pull request merged**